lib/workers/repository/update/branch/auto-replace.ts

```diff
@@ -63,35 +55,52 @@ export async function confirmIfDepUpdated(
     return false;
   }
 
-  if (newValue && newUpgrade.currentValue !== newValue) {
+  if (upgrade.newName && upgrade.newName !== newUpgrade.depName) {
+    logger.debug(
+      {
+        manager,
+        packageFile,
+        currentDepName: upgrade.depName,
+        newDepName: newUpgrade.depName,
+      },
+      'depName is not updated'
+    );
+    return false;
+  }
+
+  if (upgrade.newValue && upgrade.newValue !== newUpgrade.currentValue) {
     logger.debug(
       {
         manager,
         packageFile,
-        expectedValue: newValue,
+        expectedValue: upgrade.newValue,
         foundValue: newUpgrade.currentValue,
       },
       'Value is not updated'
     );
     return false;
   }
 
-  if (!newDigest) {
-    return true;
-  }
-  if (newUpgrade.currentDigest === newDigest) {
-    return true;
-  }
-  if (!currentDigest) {
-    if (!pinDigests) {
-      return true;
-    } else if (newDigest) {
-      return true;
+  if (
+    upgrade.updateType !== 'replacement' &&
+    (upgrade.pinDigests || upgrade.currentDigest)
+  ) {
+    if (upgrade.newDigest && upgrade.newDigest !== newUpgrade.currentDigest) {
```

Comment:
will there be another condition later or why you still left the nested if?

Reply:
Resolved with a further update and aggregation of logic